### PR TITLE
kie-issues#365: Top-left cell data is getting lost after Cutting and Pasting to the DMN Editor's Boxed Expression Editor

### LINKED ISSUE #365: Top-left cell data is getting lost after Cutting and Pasting to the DMN Editor's Boxed Expression Editor
Tested in macOS

Try to use `CMD + X` and `CMD + C` in any selection of a Decision Table. Only a subset of the selection is correctly copied.

https://github.com/kiegroup/kie-issues/assets/16005046/4cfae3dd-ccda-4113-ab56-366bf764ef17

### PR DESCRIPTION
**Pull request by yesamer**
Closes https://github.com/kiegroup/kie-issues/issues/365

The fix consists of storing the "cut" cell value before setting it with an empty value. Can't explain why the bug occurs for the first selected cell only. 

I removed lines 591 and 614 because duplicates of 589 and 612; It seems to me these are useless statements.

### CODE REVIEW

File: packages/boxed-expression-component/src/selection/BeeTableSelectionContext.tsx [hunk 1]
```diff
@@ -588,7 +588,6 @@ export function BeeTableSelectionContextProvider({ children }: React.PropsWithCh
         for (let r = startRow; r <= endRow; r++) {
           clipboardMatrix[r - startRow] ??= [];
           for (let c = startColumn; c <= endColumn; c++) {
-            clipboardMatrix[r - startRow] ??= [];
```
Comment (jomarko) [inline packages/boxed-expression-component/src/selection/BeeTableSelectionContext.tsx:591]: I struggle to find docs for `??=` operator. Is it expected we remove it just form `copy`? I still see it in `cut`.

**Review by jomarko — COMMENTED** [review]
@yesamer thank you. change seems to work. I used vsix to test manually. However I am not súre I understand `??=` operator.

File: packages/boxed-expression-component/src/selection/BeeTableSelectionContext.tsx (hunk 1 above)
Comment (yesamer) [reply packages/boxed-expression-component/src/selection/BeeTableSelectionContext.tsx:591]: @jomarko  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment
For cut, I removed it in line 614

**Review by jomarko — APPROVED** [review]

File: packages/boxed-expression-component/src/selection/BeeTableSelectionContext.tsx (hunk 1 above)
Comment (jomarko) [reply packages/boxed-expression-component/src/selection/BeeTableSelectionContext.tsx:591]: @yesamer thank you for docs link. regarding `cut`, I looked for a wrong file version probbaly.

**Review by tiagobento — APPROVED** [review]

**Pull request merged**